### DISCUSSION TITLE
Introducing PocketPages: native JS Server-Side pages for PocketBase

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#5444] [Show and tell]
Hey everyone, this project came out pretty nicely and I wanted to share it with the community. I hope it's useful to someone else. It's totally open source so feel free to try it and send PRs :D

It's called PocketPages, and it lets you code web apps PHP-style using server-side files:

```ejs
// app/index.ejs
<%= `Hello, world!` %>
```

Every file is a URL endpoint. You have full access to the PocketBase JS hooks API and all the server-side goodies, but with more traditional file-based routing.

You can try it now, it's getting stable:

```bash
bunx pocketpages new myapp --template=daisyui
cd myapp
bun dev
```

docs: https://pocketpages.dev (running on PocketPages!)
repo: https://github.com/benallfree/pocketpages

Cheers

(I'll keep this thread updated with new release info)

**Comment by ganigeorgiev** [comment]
_Side-note: I haven't checked the above implementation but note that there are plans to create something similar builtin as `pb_pages` in the future (it will be handled as part of [task#52653475](https://github.com/orgs/pocketbase/projects/2/views/1?filterQuery=pb_pages&pane=issue&itemId=52653475))._

**Reply by hookenz** [reply]
I know your needs have changed.  I would love to see pocketbase with better support for htmx style server rendered apps.  Instead of simply returning json from collections, in the UI you could add a template and attach it.  When the client requests html as the content-type it can populate the html with with the template.  

Toward something like that, how can I achieve that in middleware?

**Comment by benallfree** [comment]
PocketPages 0.0.3 has been released.

```bash
bunx pocketpages --version
```

Changes:
* `pocketpages dev` now downloads and runs PocketBase for local development
* A `+pocketpages.pb.js` bootstrap file is now automatically added to the `app/` directory, making it about as close as we can get to the "1 file" experience

**Comment by benallfree** [comment]
PocketPages 0.0.4 has been released.

```bash
bunx pocketpages --version
```

Changes:
* `pocketpages new <app>` creates a new app from a growing list of starter templates
* bug fixes and doc updates

**Comment by benallfree** [comment]
## 0.1.0

https://pocketpages.dev

### Minor Changes

- df482a4: Fix doc site and starter kit to include bootstrap files
- 1a416fe: Added `pocketbase` support in package.json for version locking
- 1a416fe: Added `init` command
- Migrated to gobot-pocketbase
- df482a4: Copy to clipboard button touch device updates
- df482a4: Fix installation instructions
- 1a416fe: Added `serve` command
- 1a416fe: Added `version` command
- df482a4: Update home screen
- 1a416fe: Removed `dev` command in favor of `serve` command

**Comment by benallfree** [comment]
## 0.2.0

### Minor Changes

- 18450fb: Added `degit` command for better template support

### Patch Changes

- 18450fb: Doc fixes for starter kits
- 18450fb: Fixed `new` command where manual names weren't properly parsed
- 18450fb: Docs for custom domains
- 18450fb: Refactored starter kits & docs

**Comment by benallfree** [comment]
## 0.3.0

### Minor Changes

- 6f82288: Added ability to only use the EJS preprocessor for specific file types (default .md and .ejs)

### Patch Changes

- 6f82288: Docs: Added Amazon SES SMTP speedrun
- 6f82288: Imroved log formatting
- 6f82288: Added support for +config.js in root
- 6f82288: Docs: added +config.js section
- 6f82288: Fix: redirect to / suffix if hitting index handler
- b27b11a: Speedrun: One-Time Passwords
- c9e2fb4: Added ca-certificates to Dockerfiles (needed for SMTP sending)
- 6f82288: Fixed static file serving bugs
- 6f82288: Enh: issue warning when loading from ./lib instead of npm package
- b27b11a: Cosmetic fixes

**Comment by benallfree** [comment]
## 0.4.0

### Minor Changes

- 0fb14db: Add one time password (OTP) support and CLI `login` command
- 0fb14db: Added support for POCKETPAGES_HOME and POCKETPAGES_MOTHERSHIP_URL env vars
- 0fb14db: Added support for local settings storage (such as auth token)

### Patch Changes

- 0fb14db: Now using @inquirer/prompts

**Comment by benallfree** [comment]
## 0.7.0

### Minor Changes

- 31baf24: Enh: Added url() parser to context
- 31baf24: Enh: asset() resolver to add cache busting when $app.isDev()

### Patch Changes

- Enh: Markdown images now use apply() for cache busting

## 0.6.0

### Minor Changes

- 699aa06: Added support for +middleware.js
- b58a02b: Migrated to pocodex
- Doc updates for 0.5.0

### Patch Changes

- 699aa06: Swapped out deprecated $app.cache()
- ea831b0: Docs: added Gmail speedrun
- dc89a4c: Implement EJS caching using $app.cache()
- dc89a4c: Fix: UTF-8 support in template engine
- ea831b0: Docs: updated SES speedrun

## 0.5.1

### Patch Changes

- 96dc569: Fixed index.md location
- 96dc569: Fix: write initial index.md to pb_hooks/pages instead of just pages
- 96dc569: Fix: throw error of pages/ doesn't exist
- 96dc569: Fix: serve static file if exists

## 0.5.0

### Minor Changes

- 2c63a44: Migrated to pocodex plugin

## 0.4.2

### Patch Changes

- Added daisyui-docs starter kit

## 0.4.1

### Patch Changes

- 1c7f643: Fix: fs-extra/esm import
